File: spring-webflux/src/main/java/victor/training/reactive/usecase/complex/ProductDetailsResponse.java
package victor.training.reactive.usecase.complex;

import lombok.Data;

@Data
public class ProductDetailsResponse {
   private Long id;
   private String name;
   private boolean active;
   private boolean resealed;

   public Product toEntity() {
      return new Product(id,name,active,resealed, null);
   }
}

File: spring-webflux/src/main/java/victor/training/reactive/usecase/complex/ProductDetailsResponse.kt
package victor.training.reactive.usecase.complex

class ProductDetailsResponse(
    var id: Long? = null,
    var name: String? = null,
    var isActive:Boolean = false,
    var isResealed:Boolean = false,
) {
    fun toEntity(): Product {
        return Product(id!!, name!!, isActive, isResealed, null)
    }

}